prover/src/constraints/evaluator/default.rs

```diff
@@ -339,3 +405,191 @@ where
         self.transition_constraints.num_aux_constraints()
     }
 }
+
+// LAGRANGE CONSTRAINTS BATCH EVALUATOR
+// ================================================================================================
+
+/// Contains a specific strategy for evaluating the Lagrange kernel boundary and transition
+/// constraints where the divisors' evaluation is batched.
+///
+/// Specifically, [`batch_inversion`] is used to reduce the number of divisions performed.
+struct LagrangeConstraintsBatchEvaluator<E: FieldElement> {
+    lagrange_kernel_boundary_constraint: Option<LagrangeKernelBoundaryConstraint<E>>,
+    lagrange_kernel_transition_constraints: Option<LagrangeKernelTransitionConstraints<E>>,
+    aux_rand_elements: AuxTraceRandElements<E>,
+}
```

Comment:
I would move this to `src/constraints/evaluator/lagrange.rs` module. Also, as mentioned in the previous comment, if we make the whole filed for `LagrangeConstraintsBatchEvaluator` optional, we won't need to make the internal fields optional.

Lastly, `aux_rand_elements` are the $r$-values, right? If so, I'd change this filed to `rand_elements: Vec<E>` as these values will be different from `AuxTraceRandElements`.